api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsManagerImpl.java

```diff
@@ -0,0 +1,1372 @@
+        return skipLoginPage;
+    }
+
+    /**
+     * @return true if the app is configured to use cookies to store the association to the Google user
+     * @since 3.0
+     */
+    @Unstable
+    public boolean isAuthWithCookies() {
+        return authWithCookies;
+    }
+
+    /**
+     * @return how long (in seconds) the cookies should be valid
+     * @since 3.0
+     */
+    @Unstable
+    long getConfigCookiesTTL() {
+        return configCookiesTTL;
+    }
+
+    private void readConfigDoc(XWikiContext context) {
+
+        try {
+            log.warn("Attempting to fetch Config doc");
+            BaseObject config = getConfigDoc(context);
+            if (config != null) {
+                configActiveFlag   = 0 != config.getIntValue("activate");
+                useCookies         = 0 != config.getIntValue("useCookies");
+                skipLoginPage      = 0 != config.getIntValue("skipLoginPage");
+                authWithCookies    = 0 != config.getIntValue("authWithCookies");
+                configAppName      = config.getStringValue("appname").trim();
+                configClientId     = config.getStringValue("clientid").trim();
+                configClientSecret = config.getStringValue("secret").trim();
+                configDomain       = config.getStringValue("domain").trim();
+                if (configDomain.length() == 0) {
+                    configDomain = null;
+                }
+                List<String> configScopes = Arrays.asList(config.getStringValue("scope").split("\\s"));
+                configScopeUseAvatar = configScopes.contains(AVATAR);
+                configScopeUseDrive = configScopes.contains("drive");
+                configCookiesTTL = config.getLongValue("cookiesTTL");
+            }
+        } catch (XWikiException e) {
+            e.printStackTrace();
+            if (log != null) {
+                log.warn("can't fetch Config doc");
+            }
+        }
+    }
+
+    /**
+     * Reads the manifest to find when the JAR file was assembled by maven.
+     * @return the build date.
+     * @since 3.0
+     */
+    @Unstable
+    public Date getBuildTime() {
+        try {
+            Class clazz = getClass();
+            String className = clazz.getSimpleName()
+                    + ".class";
+            String classPath = clazz.getResource(className).toString();
+            String manifestPath = classPath.substring(0, classPath.lastIndexOf("!") + 1)
+                    + "/META-INF/MANIFEST.MF";
+            Manifest manifest = new Manifest(new URL(manifestPath).openStream());
+            Attributes attr = manifest.getMainAttributes();
+            return new Date(Long.parseLong(attr.getValue("Bnd-LastModified")));
+        } catch (IOException e) {
+            String msg = "Can't read build time.";
+            log.warn(msg, e);
+            throw new RuntimeException(msg, e);
+        }
+    }
+
+
+    // from ActiveDirectorySetupListener
```

Comment:
We should get read of this kind of comments.

Reply:
So, the idea is to indicate where this was inspired from.
Are you sure we want to get rid of this?
paul